### ISSUE TITLE
Referencing otherwise unused variable inside catch {} strips out the variable outside try/catch, but leaves variable in catch untouched

### ISSUE THREAD

**Issue by revosw** [TanStack/router#2815]
### Which project does this relate to?

Start

### Describe the bug

_I was unsure where to file the issue, I'm guessing it's probably a bundler or other downstream process that's causing this_

In the stackblitz reproduction linked below, I have this file at `/routes/index.tsx`:
```
import { createFileRoute } from "@tanstack/react-router";

export const Route = createFileRoute("/")({
	component: RouteComponent,
});

function RouteComponent() {
	let a = undefined;
	try {
		throw "";
	} catch {
		a = 5;
	}

	return "Hello /!";
}
```
I'm creating a variable that is unused other than inside the catch{} block. It seems like the `a` variable is being stripped out when rendering the page, but the `a = 5` statement is still preserved, leading to `ReferenceError: a is not defined`

### Your Example Website or App

https://stackblitz.com/edit/github-avg9mj-mxdthn?file=app%2Froutes%2F__root.tsx,app%2Froutes%2Findex.tsx,app%2Fclient.tsx

### Steps to Reproduce the Bug or Issue

1. `npm run dev`
2. When the dev server is up and the root route has loaded, notice that the console has an error `Error in renderToPipeableStream: ReferenceError: a is not defined`

### Expected behavior

The `a` variable should not be stripped out, but instead realize that `a = 5` in the catch block relies on the `a` variable

### Platform

- OS: Windows
- Browser: Microsoft Edge
- Version: 131.0.2903.51 (Official build) (64-bit)

**Comment by SeanCassiere** [comment]
This issue seems to stem from a bug in the `bable-dead-code-elimination` package that we use for the removal of unused code.

Tracking for the issue on the `babel-dead-code-elimination` repo - https://github.com/pcattori/babel-dead-code-elimination/issues/25

**Comment by schiller-manuel** [comment]
fixed in babel-dead-code-elimination 1.0.7
